src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.Ide.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.CodingConventions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Options
+{
+    internal sealed partial class EditorConfigDocumentOptionsProvider
+    {
+        private static readonly ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>> s_projectNotifications =
```

Comment:
Why is this the key? I can guess, but the answers terrify me.

Reply:
😆 I need to document this for sure.

The CodingConventions library creates a single `CodingConventionsChangedEventArgs` when it receives a file changed notification from the watcher. It then multiple-dispatches that instance to all relevant contexts, which includes e.g. all source documents within a single folder. The map allows us to deterministically de-duplicate file system events within a batching context (one project).